### ISSUE TITLE
Grain interface has to be marked as IGrain to be implemented by a Grain.

### ISSUE THREAD

**Issue by shayhatsor** [dotnet/orleans#504]
Currently, we are trying to solve the problem of cross-cluster communication. 
What we want to do:
* Have multiple Orleans clusters in production to allow parallel development of services.
* Each Orleans cluster will represent a service that's loosely couple to others.
* Each service will have its own life-cycle and might be compiled with a different Orleans version.
* Each service will have a facade layer that will be implemented as a stateless worker and be available as a web service.
* The developer will mark the grain for web service generation.

We wanted to do something like this :
```csharp
//this is in a common dll
interface IUserService 
{ 
  Task DoIt();
}

//this is in the silo
interface IUserServiceGrain : IUserService ,IGrain {/*empty*/ }

[StatelessWorker]
class UserServiceGrain : Grain, IUserServiceGrain
{ 
  public Task DoIt(){...}
}

//this is in the service client
Task CallDoIt()
{
   var serviceClient = GetServiceClient<IUserService>();
   return serviceClient.DoIt();
}
```
but it doesn't work, since we have to have the IGrain tag on IUserService for code generation to work, and that makes the client side have to reference Orleans, which will add an unnecessary dependency.
thanks in advance for your help.

**Comment by sergeybykov** [comment]
So the issue is that codegen doesn't consider `Task DoIt()` a method on `IUserServiceGrain` interface, and doesn't make `UserServiceGrainReference` implement it, correct? Otherwise, it would just work. This came up in the past - http://orleans.codeplex.com/discussions/541405.

In theory it should be easy to have codegen consider inherited methods part of the grain interface. I don't remember if we tried and if there were any issues with that. It may be as simple as passing `bAllMethods=true` to `GrainInterfaceData.GetMethods()` - https://github.com/dotnet/orleans/blob/master/src/Orleans/CodeGeneration/GrainInterfaceData.cs#L149.

**Comment by shayhatsor** [comment]
@kumbhare, thanks! This is my take: https://github.com/dotnet/orleans/pull/515

**Comment by gabikliot** [comment]
Fixed via #515.
